CMakeModules/InternalUtils.cmake

```diff
@@ -30,7 +30,7 @@ endfunction()
 
 function(arrayfire_get_cuda_cxx_flags cuda_flags)
   if(NOT MSVC)
-    set(flags "-std=c++11 -Xcompiler -fPIC -Xcompiler ${CMAKE_CXX_COMPILE_OPTIONS_VISIBILITY}hidden")
+    set(flags "-std=c++14 -Xcompiler -fPIC -Xcompiler ${CMAKE_CXX_COMPILE_OPTIONS_VISIBILITY}hidden")
```

Comment:
are you such a change/move should nt be split into a sub separate PullRequest "upgrade to std c++14" ?

Reply:
Yeah, Thats a good idea. I can do that.